gravitee-apim-repository/gravitee-apim-repository-elasticsearch/src/test/java/io/gravitee/repository/elasticsearch/spring/ElasticsearchRepositoryConfigurationTest.java

```diff
@@ -34,20 +57,111 @@
 @Import(ElasticsearchRepositoryConfiguration.class)
 public class ElasticsearchRepositoryConfigurationTest {
 
+    public static final String ELASTICSEARCH_VERSION = "7.17.8";
+    public static final String CLUSTER_NAME = "gravitee_test";
+
+    @Value("${elasticsearch.version:" + ELASTICSEARCH_VERSION + "}")
+    private String elasticsearchVersion;
+
     @Bean
     public Vertx vertx() {
         return Vertx.vertx();
     }
 
     @Bean
-    public ElasticsearchNode elasticsearchNode() {
-        return new ElasticsearchNode();
+    public Client client(RepositoryConfiguration repositoryConfiguration) throws InterruptedException {
+        HttpClientConfiguration clientConfiguration = new HttpClientConfiguration();
+        clientConfiguration.setEndpoints(repositoryConfiguration.getEndpoints());
+        clientConfiguration.setUsername(repositoryConfiguration.getUsername());
+        clientConfiguration.setPassword(repositoryConfiguration.getPassword());
+        clientConfiguration.setRequestTimeout(60_000);
+        return new HttpClient(clientConfiguration);
+    }
+
+    @Bean
+    public String indexSampleData(Client client, FreeMarkerComponent freeMarkerComponent) throws InterruptedException {
```

Comment:
why creating a bean returning a string which does not seem to be used ?

Reply:
I've used the bean creation mechanism to insert data into my Elastic container only once.
But maybe there is a easier and/or safer way to do that